### ISSUE TITLE
`case Callable()` and match exhaustiveness

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#8160]
**Describe the bug**

matching against `x: Callable[[], int]` with `case Callable()` does not make exhaustive match.

**Code or Screenshots**

```python
from collections.abc import Callable

def f(x: Callable[[], int]):
    if isinstance(x, Callable):
        return x()

reveal_type(f)

def g(x: Callable[[], int]):
    match x:
        case Callable():
            return x()

reveal_type(g)
```

```
cc.py
  cc.py:7:13 - information: Type of "f" is "(x: () -> int) -> int"
  cc.py:14:13 - information: Type of "g" is "(x: () -> int) -> (int | None)"
0 errors, 0 warnings, 2 informations
```

`g` should have type `(x: () -> int) -> int`

**VS Code extension or command-line**

command-line, 1.1.367

**Comment by erictraut** [comment]
Thanks for the bug report. This is addressed in pyright 1.1.368.
